src/mem/ruby/protocol/MOESI_AMD_Base-dir.sm

```diff
@@ -527,9 +533,11 @@ machine(MachineType:Directory, "AMD Baseline protocol")
         out_msg.ForwardRequestTime := tbe.ForwardRequestTime;
         out_msg.ProbeRequestStartTime := tbe.ProbeRequestStartTime;
         out_msg.OriginalResponder := tbe.LastSender;
-        if(tbe.atomicData){
-          out_msg.WTRequestor := tbe.WTRequestor;
-        }
+        out_msg.isGLCSet := tbe.isGLCSet;
+        out_msg.isSLCSet := tbe.isSLCSet;
+	if(tbe.atomicData){
```

Comment:
@mattsinc is this the part that you wanted review on? I think this should be fine, tbe.atomicData flag is set on an atomic request, irrespective of whether return data is needed. Though, looking through this, I imagine that we would not need tbe.atomicDataReturn and tbe.atomicDataNoReturn flags anymore since we can just use the GLC bits (since I believe they're now being propagated through all cache levels).

Reply:
Yes, this is it.  I don't know if I agree about GLC/SLC vs. atomicDataReturn/NoReturn though -- there can be GLC atomics that either return or don't return, for example.

In any event I don't think this involves a decision for this patch though.